### DISCUSSION TITLE
贡献 xrepo 增加新配置，有没有操作规范？

### DISCUSSION THREAD

**Discussion by fuzhouch** [xmake-io/xmake#4608] [Q&A]
前一段时间作项目和 openal 交流过一段，发现 xrepo 上的默认配置开得选项太多，导致不必要的依赖关系被引入。这些可以增加一些选项来关闭。我在本地跑了一段时间自己的修改，还算顺利。感觉可以贡献回社区，但发现找不到 xrepo 发 PR 应该遵循的指导原则，包括但不限于：

1. Xrepo 的修改应该在哪个 branch 上（master， dev，或者别的，以配合 CI？）
2. PR 发送有没有要求（标题、内容说明、本地测试等）
3. 其他需要的准备内容（比如文档？）

我们有没有 wiki 来记录这些开发规范？如果没有，我也可以写一个，只要确定应该写到哪里就行。

**Comment by waruqi** [comment]
有的，可以看下这里 https://github.com/xmake-io/xmake-repo/blob/dev/CONTRIBUTING.md

一般包默认约定最小化依赖配置，用户需要再自己配置开启，当然也会有一些包，贡献者在贡献进来的时候，可能没有全部处理好这些依赖，导致默认开启了，这种可以提 pr 过来改进下
